holidays/calendars/gregorian.py

```
@@ -61,6 +61,34 @@ def _get_nth_weekday_from(n: int, weekday: int, from_dt: date) -> date:
+        month += 1
+        if month > 12:
+            month = 1
+            year += 1
+        n += 1
+
+    start_date = date(year, month, 1)
+    dt = _timedelta(start_date, -start_date.weekday() + (n - 1) * 7)
+    # If the last day of the specified week falls in another month,
+    # it's already the first (last) week of that month.
+    if _timedelta(dt, +6).month != orig_month:
+        raise ValueError(f"Month {orig_month} of {orig_year} does not have week {orig_n}")
```

Ruff (0.15.9)

[warning] Avoid specifying long messages outside the exception class

(TRY003)